action-api/src/test/java/com/powsybl/action/EqualsActionTest.java

```diff
@@ -0,0 +1,472 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package com.powsybl.action;
+
+import com.google.common.testing.EqualsTester;
+import com.powsybl.iidm.network.PhaseTapChanger;
+import com.powsybl.iidm.network.StaticVarCompensator;
+import com.powsybl.iidm.network.ThreeSides;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+/**
+ * @author Pauline JEAN-MARIE {@literal <pauline.jean-marie at artelys.com>}
+ */
+class EqualsActionTest {
+
+    @Test
+    void switchAction() {
+        SwitchAction action1 = new SwitchActionBuilder().withId("id")
+                .withNetworkElementId("neId")
+                .withOpen(true)
+                .build();
+        SwitchAction action2 = new SwitchActionBuilder().withId("id")
+                .withNetworkElementId("neId")
+                .withOpen(true)
+                .build();
+        SwitchAction action3 = new SwitchActionBuilder().withId("id2")
+                .withNetworkElementId("neId")
+                .withOpen(true)
+                .build();
+        SwitchAction action4 = new SwitchActionBuilder().withId("id")
+                .withNetworkElementId("neId2")
+                .withOpen(true)
+                .build();
+        SwitchAction action5 = new SwitchActionBuilder().withId("id")
+                .withNetworkElementId("neId")
+                .withOpen(false)
+                .build();
+        new EqualsTester()
+                .addEqualityGroup(action1, action2)
+                .addEqualityGroup(action3)
+                .addEqualityGroup(action4)
+                .addEqualityGroup(action5)
+                .testEquals();
+    }
+
+    @Test
+    void terminalConnectionAction() {
+        Action action1 = new TerminalsConnectionActionBuilder().withId("id")
+                .withNetworkElementId("neId")
+                .withOpen(true)
+                .withSide(ThreeSides.ONE)
+                .build();
+        TerminalsConnectionAction action2 = new TerminalsConnectionActionBuilder().withId("id")
+                .withNetworkElementId("neId")
+                .withOpen(true)
+                .withSide(ThreeSides.ONE)
+                .build();
+        Action action3 = new TerminalsConnectionActionBuilder().withId("id")
+                .withNetworkElementId("neId")
+                .withOpen(true)
+                .build();
+        Action action4 = new SwitchActionBuilder().withId("id")
+                .withNetworkElementId("neId")
+                .withOpen(true)
+                .build();
+        TerminalsConnectionAction action5 = new TerminalsConnectionActionBuilder().withId("id")
```

Comment:
To fully cover the `equals()` method, could you add actions with changes on the following attributes:
- open
- networkElementId